### PR TITLE
Added e2e testing for multiplex flows

### PR DESCRIPTION
**Pull request by johanna-skylight**
# FRONTEND PULL REQUEST

## Related Issue

Implements #4997 

## Changes Proposed

- Add a new spec 09 to cover multiplex testing and includes: created a multiplex device, add device to facility and conduct a test with the multiplex device

## Additional Information

- A new command was added to cypress to allow for manual post request.
- Post request were added as pre steps for the tests to ensure the data is well set before running them.

## Testing
- Setup the new variable CYPRESS_BACKEND_URL in cypress.env.json
- Successfully run spec 09 in local

## Screenshots / Demos

- For large changes, please pair with a designer to ensure changes are as intended

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -116,3 +117,18 @@
...
+        method: 'POST',
+        url: graphqlURL,
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        body: requestBody
+      }
+    ))
+  )
+})
```
[the remark this answers is not in the document]
Comment (emmastephenson) [reply cypress/support/commands.js:134]: @johanna-skylight  Just want to make sure you saw this semicolon check!

File: cypress/utils/graphql-test-utils.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Utility to match GraphQL mutation based on the operation name
+export const hasOperationName = (req, operationName) => {
```
Comment (emmastephenson) [inline cypress/utils/graphql-test-utils.js:2]: Thanks for adding these utilities!

**Review by emmastephenson — CHANGES_REQUESTED** [review]
Looking good, thanks for the through e2e test! I'm not sure how much of that device flow has been tested, so this is great. Just a couple of minor changes requested.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java [hunk 3]
```diff
@@ -27,4 +28,20 @@ public DeviceType updateDeviceType(@Argument UpdateDeviceType input)
       throws IllegalGraphqlArgumentException {
     return deviceTypeService.updateDeviceType(input);
   }
+
+  @MutationMapping
+  public DeviceType markDeviceTypeAsDeleted(@Argument UUID deviceId, @Argument String deviceName)
```
[the remark this answers is not in the document]
Comment (BobanL) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java:33]: I think `removeDeviceType` actually just uses the delete from [EternalAuditedEntityRepository](https://github.com/CDCgov/prime-simplereport/blob/63f0009b43ee791095f444ea034baf72faf0df1c/backend/src/main/java/gov/cdc/usds/simplereport/db/repository/EternalAuditedEntityRepository.java#L29-L30), which is a soft delete.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java [hunk 4]
```diff
@@ -27,4 +28,20 @@ public DeviceType updateDeviceType(@Argument UpdateDeviceType input)
...
+      throws IllegalGraphqlArgumentException {
+    DeviceType device =
+        (deviceId != null)
+            ? deviceTypeService.getDeviceType(deviceId)
+            : deviceTypeService.getDeviceType(deviceName);
+
+    if (device == null) {
+      return null;
+    }
```
Comment (BobanL) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java:42]: I tested this locally, and I don't think this null check is necessary! `removeDeviceType` does nothing if the value is null.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java:33]: yeah, I can confirm!. When testing this in local the device never got deleted and the `is_deleted` column was changed to true

File: cypress/support/commands.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply cypress/support/commands.js]: added!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java (hunk 4 above)
Comment (johanna-skylight) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java]: Nice! I removed the conditional but left the unit test for the scenario

File: backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeMutationResolver.java:33]: Ahhh thank you both, that makes more sense! 😄 I thought the delete was coming directly from JPA, thanks for checking me :)

File: cypress/integration/09-multiplex_testing_spec.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      cy.intercept('POST', graphqlURL, (req) => {
+        aliasQuery(req, 'GetFacilityQueue')
+        aliasQuery(req, 'GetPatientsByFacilityForQueue')
+        aliasQuery(req, 'EditQueueItem')
+        aliasMutation(req, 'SubmitQueueItem')
+      })
+
+      // remove a test for the patient if it exists
+      cy.makePOSTRequest({
+        "operationName": "RemovePatientFromQueue",
```
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply cypress/integration/09-multiplex_testing_spec.js:153]: it just returns a 200 with a body of removedPatient = null

File: backend/src/test/java/gov/cdc/usds/simplereport/service/DeviceTypeServiceTest.java [hunk 6]
```diff
@@ -32,6 +40,21 @@ class DeviceTypeServiceTest extends BaseServiceTest<DeviceTypeService> {
   @Autowired private DeviceTypeRepository _deviceTypeRepo;
   @Autowired private SpecimenTypeRepository specimenTypeRepository;
 
+  @Mock private DeviceTypeRepository _deviceTypeRepoMock;
+
+  private DeviceTypeService deviceTypeServiceWithMock;
+
+  @BeforeEach
```
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/DeviceTypeServiceTest.java]: I changed it for a BeforeAll

File: backend/src/test/java/gov/cdc/usds/simplereport/api/graphql/DeviceManagementTest.java [hunk 7]
```diff
@@ -134,4 +144,39 @@ private Map<String, Object> getUpdateDeviceTypeInput(UUID internalId) {
 
     return Map.of("input", input);
   }
+
+  @Test
+  void markDeviceTypeAsDeleted_uuid_success() {
```
Comment (emmastephenson) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/graphql/DeviceManagementTest.java:149]: Thank you for adding the mutation tests here! Definitely a good habit to get into

**Review by emmastephenson — APPROVED** [review]
LGTM, this looks great! Thanks for extending our test coverage! 😄

**Review by BobanL — APPROVED** [review]
Looks great! Thank you :)

**Pull request merged**